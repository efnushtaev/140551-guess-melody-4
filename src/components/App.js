/* eslint-disable react/prop-types */
import React from 'react';
import WelcomeScreen from './welcome-screen';

const App = ({settings}) => {
  const {gameTime, errorCount} = settings;
  return <WelcomeScreen
    gameTime={gameTime}
    errorCount={errorCount}/>;
};

export default App;
