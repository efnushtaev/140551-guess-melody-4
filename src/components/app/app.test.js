/* eslint-disable no-alert */
import React from 'react';
import renderer from 'react-test-renderer';
import App from './app';
import {describe, expect, it} from '@jest/globals';

const Settings = {
  GAME_TIME: 3,
  ERROR_COUNTING: 5
};

const questions = [{
  type: `genre`,
  genre: `indie`,
  answers: [
    {
      genre: `hard`,
      src: `https://upload.wikimedia.org/wikipedia/commons/5/53/BWV_543-prelude.ogg`
    },
    {
      genre: `blues`,
      src: `https://upload.wikimedia.org/wikipedia/commons/5/53/BWV_543-prelude.ogg`
    },
    {
      genre: `indie`,
      src: `https://upload.wikimedia.org/wikipedia/commons/5/53/BWV_543-prelude.ogg`
    },
    {
      genre: `jazz`,
      src: `https://upload.wikimedia.org/wikipedia/commons/5/53/BWV_543-prelude.ogg`
    },
  ]
},
{
  type: `artist`,
  title: `Кто исполняет эту песню?`,
  src: `https://upload.wikimedia.org/wikipedia/commons/5/53/BWV_543-prelude.ogg`,
  answers: [
    {
      artist: `Пелагея`,
      picture: `https://smk.org.uk/wp-content/uploads/avatar.jpg`
    },
    {
      artist: `Краснознамённая дивизия имени моей бабушки`,
      picture: `https://smk.org.uk/wp-content/uploads/avatar.jpg`
    },
    {
      artist: `Lorde`,
      picture: `https://smk.org.uk/wp-content/uploads/avatar.jpg`
    }]
}
];

describe(`appTest`, () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(<App
        questions={questions}
        settings={Settings}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
