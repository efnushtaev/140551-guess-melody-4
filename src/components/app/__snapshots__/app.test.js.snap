// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appTest renders correctly 1`] = `
<section
  className="game game--genre"
>
  <div
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </div>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer0"
          name="answer0"
          onChange={[Function]}
          type="radio"
        />
        <label
          className="artist__name"
          htmlFor="answer0"
        >
          <img
            className="artist__picture"
          />
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer1"
          name="answer1"
          onChange={[Function]}
          type="radio"
        />
        <label
          className="artist__name"
          htmlFor="answer1"
        >
          <img
            className="artist__picture"
          />
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer2"
          name="answer2"
          onChange={[Function]}
          type="radio"
        />
        <label
          className="artist__name"
          htmlFor="answer2"
        >
          <img
            className="artist__picture"
          />
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer3"
          name="answer3"
          onChange={[Function]}
          type="radio"
        />
        <label
          className="artist__name"
          htmlFor="answer3"
        >
          <img
            className="artist__picture"
          />
        </label>
      </div>
    </form>
  </section>
</section>
`;
